### PR TITLE
Proto array fork choice tree handler

### PR DESCRIPTION
**Pull request by terencechain**
This PR implements a tree handler for proto array fork choice. It dumps out the current block tree in GraphViz DOT Format. How to use it:

1.) Enable proto array fork choice `--proto-array-forkchoice`
2.) Goto `http://localhost:8080/tree`
3.) Copy the digraph and paste it to https://dreampuf.github.io/GraphvizOnline/ or http://www.webgraphviz.com/
4.) Exported fields are node's `bestDescendant`, `weight`, `slot` and `parent`

Example:
Screen Shot 2020-01-26 at 9 52 09 AM

### CODE REVIEW

File: beacon-chain/forkchoice/interfaces.go [hunk 1]
```diff
@@ -2,6 +2,7 @@ package forkchoice
 
 import (
 	"context"
+	"github.com/prysmaticlabs/prysm/beacon-chain/forkchoice/protoarray"
```
Comment (0xKiwi) [inline beacon-chain/forkchoice/interfaces.go]:
```suggestion

	"github.com/prysmaticlabs/prysm/beacon-chain/forkchoice/protoarray"
```

File: beacon-chain/forkchoice/interfaces.go [hunk 2]
```diff
@@ -10,6 +11,7 @@ type ForkChoicer interface {
 	BlockProcessor       // to track new block for fork choice.
 	AttestationProcessor // to track new attestation for fork choice.
 	Pruner               // to clean old data for fork choice.
+	Getter               // to retrieve fork choice information
```
Comment (0xKiwi) [inline beacon-chain/forkchoice/interfaces.go]:
```suggestion
	Getter               // to retrieve fork choice information.
```

File: tools/cluster-pk-manager/server/db.go [hunk 3]
```diff
@@ -348,4 +348,3 @@ func (d *db) RemovePKFromPod(podName string, key []byte) error {
 		return tx.Bucket(assignedPkBucket).Put([]byte(podName), marshaled)
 	})
 }
```
Comment (0xKiwi) [inline tools/cluster-pk-manager/server/db.go]:
```suggestion
}

```

**Comment by handelaar2** [comment]
Ran the tree through webgraphviz.com. Don't understand why one branch shows bestDescendent=-1 where it looks it should be 49. What am I missing?

Screenshot 2020-01-27 at 09 04 19

**Pull request merged**